packages/core/src/node/logger-cli-contribution.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as yargs from 'yargs';
+import { injectable } from 'inversify';
+import { LogLevel } from '../common/logger';
+import { CliContribution } from './cli';
+import * as fsextra from 'fs-extra';
+import * as nsfw from 'nsfw';
+import { Event, Emitter } from '../common/event';
+import * as path from 'path';
+
+/** Maps logger names to log levels.  */
+export interface LogLevels {
+    [key: string]: LogLevel,
+}
+
+/**
+ * Parses command line switches related to log levels, then watches the log
+ * levels file (if specified) for changes.  This is the source of truth for
+ * what the log level per logger should be.
+ */
+@injectable()
+export class LogLevelCliContribution implements CliContribution {
+    protected _logLevels: LogLevels = {};
+
+    /**
+     * Log level to use for loggers not specified in `logLevels`.
+     */
+    protected _defaultLogLevel: LogLevel = LogLevel.INFO;
+
+    protected logLevelsChangeEvent: Emitter<void> = new Emitter<void>();
+
+    get defaultLogLevel(): LogLevel {
+        return this._defaultLogLevel;
+    }
+
+    get logLevels(): LogLevels {
+        return this._logLevels;
+    }
+
+    configure(conf: yargs.Argv): void {
+        conf.option('log-level', {
+            description: 'Sets the default log level',
+            choices: Array.from(LogLevel.strings.values()),
+            nargs: 1,
+        });
+
+        conf.option('log-levels-file', {
+            description: 'Path to a JSON file specyfing the log level of various loggers',
+            type: 'string',
+            nargs: 1,
+        });
+    }
+
+    async setArguments(args: yargs.Arguments): Promise<void> {
+        if (args['log-level'] !== undefined && args['log-levels-file'] !== undefined) {
+            throw new Error('--log-level and --log-levels-file are mutually exclusive.');
+        }
+
+        if (args['log-level'] !== undefined) {
+            this._defaultLogLevel = this.readLogLevelString(args['log-level'], 'Unknown log level passed to --log-level');
+        }
+
+        if (args['log-levels-file'] !== undefined) {
```

Comment:
How about `log-config`? I think its format will change overtime to support more complex configurations, not only levels, e.g. https://github.com/theia-ide/theia/issues/1687

Reply:
Done.